src/system/Icons.cpp

```diff
@@ -32,9 +34,30 @@ static std::optional<std::string> readFileAsString(const std::string& path) {
     return trim(std::string((std::istreambuf_iterator<char>(file)), (std::istreambuf_iterator<char>())));
 }
 
-static const std::array<const char*, 3> ICON_THEME_DIRS = {"/usr/share/icons", "/usr/local/share/icons", "~/.icons"};
+static std::vector<std::string> getIconSearchPaths() {
+  std::vector<std::string> paths;
+
+  // user icons (e.g., ~/.local/share/icons)
+  const gchar* userDataDir = g_get_user_data_dir();
+  if (userDataDir) {
+    paths.push_back(std::string(userDataDir) + "/icons");
+  }
+
+  // system icons (e.g. /usr/local/share/icons, /usr/share/icons)
+  const gchar* const* systemDataDirs = g_get_system_data_dirs();
```

Comment:
these can be trivially hardcoded as static / constexpr arrays of path suffixes, we do that everywhere else.